### ISSUE TITLE
[ADD EXAMPLE] Find the intersection of two lines

### ISSUE THREAD

**Issue by jeremydouglass** [jeremydouglass/rosetta_examples_p5#35]
### URL of task
https://rosettacode.org/wiki/Find_the_intersection_of_two_lines

### Next steps
- [x] I will write the .pde sketch and submit as a pull request

**Comment by villares** [comment]
Hi @jeremydouglass!
I had some code for that...  so I couldn't resist fiddling and porting it from Python to Java!
I hope I'm not duplicating or being pushy...
Could you have a look if my code is OK at the wiki?

...also the pure Python Rosetta example is so ...unpythonic! but I feel bad about changing it.

**Comment by villares** [comment]
I couldn't help but I changed the Python 3.7 version (it's a wiki right?)
...let them revert if they don't like it.

**Comment by jeremydouglass** [comment]
It is a wiki -- I really don't think you need to feel bad about making changes. Making the Python examples pythonic is exactly what Rosetta Code is all about, so go for it!

When there is a doubt about which is better / more useful, list multiple examples under a language.

...and it looks like Hout did just that -- restored the much longer version with "reduce" alongside yours.

**Comment by jeremydouglass** [comment]
See discussion here:

https://github.com/jeremydouglass/rosetta_examples_p5/issues/36

I think that this code has the same problem -- it returns null (not intersecting) if d=0. However, we can see that d=0 includes any case where the two line segments are the same, as well as any other cases where the slopes are the same (partial overlapping, or one line being a sub-segment of the other line).

```
float d = ((By2 - By1) * (Ax2 - Ax1) - (Bx2 - Bx1) * (Ay2 - Ay1));
```

So, for example, if one line is 0,0->100,100, and the other line is either the same or is 100,100->0,0, then d==0.

**Comment by villares** [comment]
I see what you mean. 

As I was going to modify the code, it struck me that even if they do 'collide' or 'overlap' there is no single intersection point (edge case would be consecutive segments...). So for these cases one couldn't return a PVector, then... null.

In short, null not as not intersecting, but as, 'I can't provide a single point'.
(could test for overlapping endpoints to be complete...)

**Comment by jeremydouglass** [comment]
That is true. Even if you wanted to just return some point, any point, if the lines are coincident you cannot return

`(x1 + (uA * (x2-x1)),  y1 + (uA * (y2-y1))`

...because this will always return x1, y1 -- even in cases where the two lines are partially overlapping and x1,y1 falls outside the overlap area.

If you needed to return some point, you _could_ just test the first two points with point-line:

```
If (x1,y1) is on line2, return it
  else if (x2,y2) is on line2, return it
else return (x3,y3) // ...because the only other case is that line 2 is a segment on line 1.
```

**Comment by villares** [comment]
I was revisiting this... could we add the version as it is for now (null & None for collinear lines, even if overlapping)?
